tests/src/test/spark330/scala/org/apache/spark/sql/rapids/suites/RapidsStringExpressionsSuite.scala

```diff
@@ -19,7 +19,38 @@
 spark-rapids-shim-json-lines ***/
 package org.apache.spark.sql.rapids.suites
 
-import org.apache.spark.sql.catalyst.expressions.StringExpressionsSuite
+import org.apache.spark.sql.catalyst.expressions.{Literal, StringExpressionsSuite, SubstringIndex}
 import org.apache.spark.sql.rapids.utils.RapidsTestsTrait
+import org.apache.spark.sql.types.StringType
 
-class RapidsStringExpressionsSuite extends StringExpressionsSuite with RapidsTestsTrait {}
+class RapidsStringExpressionsSuite extends StringExpressionsSuite with RapidsTestsTrait {
+  test("string substring_index function in rapids") {
```

Comment:
Question: Why do we need this case? Is this modified from a Spark UT case or is it a new added case? If it's new, I think we can add it to our Scala test part.